pandas/tests/test_algos.py

```diff
@@ -962,6 +962,52 @@ def test_value_counts_uint64(self):
         if not compat.is_platform_32bit():
             tm.assert_series_equal(result, expected)
 
+    def test_value_counts_nonsorted_single_occurance(self):
+        # GH 12679
+        # All items occour exactly once.
+        # No matter if sorted or not, the resulting values should be in
+        # the same order.
+        s = Series(list('bacdef'))
+
+        # Guarantee the same index if value_counts(sort=False) is used
+        vc = s.value_counts(sort=False, ascending=False)
+        tm.assert_series_equal(Series(vc.index), s)
+        vc = s.value_counts(sort=False, ascending=True)
+        tm.assert_series_equal(Series(vc.index), s)
+
+    @pytest.mark.xfail(reason="sort=True does not guarantee the same order")
+    def test_value_counts_sorted_single_occurance(self):
+        # GH 12679
+        s = Series(list('bacdef'))
+        # Guarantee does not hold yet for the sort=True case
+        vc = s.value_counts(sort=True, ascending=False)
+        tm.assert_series_equal(Series(vc.index), s)
+        vc = s.value_counts(sort=True, ascending=True)
+        tm.assert_series_equal(Series(vc.index), s)
+
+    def test_value_counts_nonsorted_double_occurance(self):
```

Comment:
parametrize these.

Reply:
The `double_occurance` tests have different expected results and I wouldn't parametrize it due to that. Or would you do this as well in that case?

Reply:
I c, ok then, parameterize over ascending though